strawberry_django/subscriptions/publishers.py

```diff
@@ -0,0 +1,110 @@
+from asgiref.sync import sync_to_async
+from channels.db import database_sync_to_async
+from django.core.exceptions import ValidationError
+from django.db.models import Model
+from strawberry.relay.types import GlobalID
+from strawberry.relay.utils import from_base64
+from strawberry.types import Info
+
+from strawberry_django.auth.utils import get_current_user
+
+from .utils import get_group, get_msg, get_msg_type
+
+
+class ModelInstanceSubscribePublisher:
+    """Publishing updated instances to graphql subscriptions.
+
+    It expects to be called with the Info context from strawberry django,
+    the instance pk and the model.
+
+    It can be used like so:
+    ```
+    publisher = ModelInstanceSubscribePublisher(info=info, pk=pk, model=model)
+    async for msg in publisher.await_messages():
+        yield msg
+    ```
+
+    """
+
+    def __init__(self, info: Info, pk: GlobalID, model: Model):
+        self.info = info
+        self.pk = pk
+        self.model = model
+
+        self.ws = info.context["ws"]
+        self.channel_layer = self.ws.channel_layer
+
+    async def verify_logged_in(self):
+        user = get_current_user(self.info)
+
+        if user.is_anonymous:
+            raise ValidationError("Anynomous users are not allowed to subscribe.")
```

Comment:
Hrm, why not?

I mean, which user is allowed to subscribe should be defined by the application. Maybe it is an application which allows that?

What do you think about adding a way to check for permissions using the model_subscriber function? We can even try to use the permissioning system we already have for this (it might need some adjustments)

Reply:
This is from the OneSila project, most endpoints are not available unless logged in.  I left it here as a reminder to investigate if we can leverage the existing extensions somehow.  Would prefer to stay away from a second set of permissions. 

Having said that, I also still want to reshuffle a few things in order to ensure the Publisher class is easy to extend and override for custom behaviour on the Subscription class.